### PR TITLE
Add cogdl

### PR DESCRIPTION
**Pull request by eriknw**
I was a bit verbose/explicit with dependencies.  I base them on what is imported by modules within the package.

`dgl` could be added as a dependency if it's added to conda-forge.  See dmlc/dgl#1855 and conda-forge/staged-recipes#12552

Even though this is a pure Python package, I run a few of the fastest tests that exercise different files, because
this package has some heavy dependencies (pytorch, pytorch_geometric, etc) and I want that warm fuzzy feeling :)

I get the tarball from github instead of PyPI, because the PyPI tarball does not (yet) contain the license file or the tests.
Currently, the github tarball is not that much larger than from PyPI.

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

CC @cenyk1230, who may be interested in this.  Would you like to be added as a maintainer or can you recommend anyone who would?

### CODE REVIEW

File: recipes/cogdl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    node classification, link prediction and other tasks on graphs. It provides
+    implementations of many popular models, including non-GNN Baselines like
+    Deepwalk, LINE, NetMF, GNN Baselines like GCN, GAT, GraphSAGE.
+  doc_url: https://keg.cs.tsinghua.edu.cn/cogdl/
+  dev_url: https://github.com/THUDM/cogdl
+
+extra:
+  recipe-maintainers:
+    - eriknw
+    - paul-tqh-nguyen
```
Comment (eriknw) [inline recipes/cogdl/meta.yaml:83]: @paul-tqh-nguyen, are you okay with being a maintainer?

Comment (paul-tqh-nguyen) [reply recipes/cogdl/meta.yaml:83]: Yes.

**Review by paul-tqh-nguyen — APPROVED** [review]

**Comment by eriknw** [comment]
@conda-forge/help-python This is ready for review.

File: recipes/cogdl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/THUDM/cogdl/archive/{{ version }}.tar.gz
+  sha256: 4c35bcc20e9bd7e9928c6e5a02d2ca974d42ad47255ad7dafd70cbec5da079d6
+
+build:
+  skip: true  # [win]
```
Comment (dopplershift) [inline recipes/cogdl/meta.yaml:13]: Can you go ahead and put this back to `noarch: python`? We don't need all the checks to pass to merge, and it *drastically* reduces the package build load.

Looks like the only windows problem is that some deps aren't available? In that case, it just won't be installable on Windows; and if they ever appear, this package will "just work".

Comment (eriknw) [reply recipes/cogdl/meta.yaml]: Done!  Thanks for taking a look @dopplershift

File: recipes/cogdl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  sha256: 4c35bcc20e9bd7e9928c6e5a02d2ca974d42ad47255ad7dafd70cbec5da079d6
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
```
Comment (dopplershift) [inline recipes/cogdl/meta.yaml:19]: Should also have `python >=3.6` here too

Comment (eriknw) [reply recipes/cogdl/meta.yaml]: Done.

**Review by dopplershift — APPROVED** [review]

**Comment by dopplershift** [comment]
Windows fail expected due to missing dependencies--ok for noarch.

**Pull request merged**